Sources/Controllers/Map/OAMapViewController.mm

```diff
@@ -1480,6 +1482,9 @@ - (void) zoomInGestureDetected:(UITapGestureRecognizer *)recognizer
     // Get base zoom delta
     float zoomDelta = [self currentZoomInDelta];
     
+    _zoomInTargetChanged = YES;
+    [self storeTargetPosition:recognizer];
```

Comment:
It seems `storeTargetPosition` / `restorePreviousTarget` should be called in `zoomInGestureDetected` method depending on recognizer.state, like in `moveGestureDetected` (`UIGestureRecognizerStateBegan` / `UIGestureRecognizerStateEnded` / `UIGestureRecognizerStateCancelled`)

Reply:
It would be good, but I tested it, and it is not true in this case. moveGestureDetected is called when we touch and remove our finger from the screen and has different states. zoomInGestureDetected is called only once if we double-click and has the UIGestureRecognizerStateChanged state.